functions/check_availability.pp

```diff
@@ -0,0 +1,30 @@
+#
+# @summary check if a group of targets are reachable for bolt
+#
+# @param targets list of targets that are going to be checked
+# @param output_details flag to enable/disable error output for failed nodes
+#
+# @return counter for unavailable nodes
+#
+# @author Tim Meusel <tim@bastelfreak.de>
+#
+function peadm::check_availability(
+  TargetSpec $targets,
+  Boolean $output_details = true
+) >> Integer {
+  $check_result = wait_until_available($targets, wait_time => 2, _catch_errors => true)
+  if $check_result.ok {
+  } else {
+    $messages = $check_result.error_set.map |$result| {
+      "## node ${result.target} has connection error '${result.error.kind}' with message '${result.error.msg}'"
```

Comment:
I'd better use `details` field of the `fail_plan()` to return list of the failed nodes (or their Result's maybe event)